x-pack/plugin/mapper-unsigned-long/src/yamlRestTest/resources/rest-api-spec/test/70_time_series.yml

```diff
@@ -0,0 +1,124 @@
+setup:
+  - skip:
+      version: " - 8.0.99"
+      reason: _tsid introduced in 8.1.0
+
+  - do:
+      indices.create:
+        index: test
+        body:
+          settings:
+            index:
+              mode: time_series
+              routing_path: [metricset]
+              time_series:
+                start_time: 2021-04-28T00:00:00Z
+                end_time: 2021-04-29T00:00:00Z
+          mappings:
+            properties:
+              "@timestamp":
+                type: date
+              metricset:
+                type: keyword
+                time_series_dimension: true
+              ul:
+                type: unsigned_long
+                time_series_dimension: true
+
+  - do:
+      bulk:
+        refresh: true
+        index: test
+        body:
+          - '{"index": {}}'
+          - '{"@timestamp": "2021-04-28T18:35:24.467Z", "metricset": "aa", "voltage": 7.2, "ul": 9223372036854775807}'
+          - '{"index": {}}'
+          - '{"@timestamp": "2021-04-28T18:35:34.467Z", "metricset": "aa", "voltage": 7.6, "ul": 9223372036854775807}'
+          - '{"index": {}}'
+          - '{"@timestamp": "2021-04-28T18:35:44.467Z", "metricset": "aa", "voltage": 7.1, "ul": 18446744073709551614}'
+          - '{"index": {}}'
+          - '{"@timestamp": "2021-04-28T18:35:54.467Z", "metricset": "aa", "voltage": 7.3, "ul": 18446744073709551615}'
+          - '{"index": {}}'
+          - '{"@timestamp": "2021-04-28T18:35:24.467Z", "metricset": "aa", "voltage": 3.2, "ul": 9223372036854775808}'
+          - '{"index": {}}'
+          - '{"@timestamp": "2021-04-28T18:35:34.467Z", "metricset": "aa", "voltage": 3.6, "ul": 9223372036854775808}'
+          - '{"index": {}}'
+          - '{"@timestamp": "2021-04-28T18:35:44.467Z", "metricset": "aa", "voltage": 3.1, "ul": 9223372036854775808}'
+          - '{"index": {}}'
+          - '{"@timestamp": "2021-04-28T18:35:54.467Z", "metricset": "aa", "voltage": 3.3, "ul": 9223372036854775807}'
+
+---
+fetch the _tsid:
+  - skip:
+      version: " - 8.00.99"
+      reason: _tsid support introduced in 8.1.0
+
+  - do:
+      search:
+        index: test
+        body:
+          fields: [_tsid, metricset, ul]
+          query:
+            query_string:
+              query: '+@timestamp:"2021-04-28T18:35:24.467Z" +metricset:aa'
+
+  - match: {hits.total.value: 2}
+  - match: {hits.hits.0.fields._tsid: [{metricset: aa, ul: 9223372036854775808}]}
+  - match: {hits.hits.0.fields.metricset: [aa]}
+  - match: {hits.hits.0.fields.ul: [9223372036854775808]}
+  - match: {hits.hits.1.fields._tsid: [{metricset: aa, ul: 9223372036854775807}]}
+  - match: {hits.hits.1.fields.metricset: [aa]}
+  - match: {hits.hits.1.fields.ul: [9223372036854775807]}
+
+---
+# TODO: Sort order is wrong here. Probably this is caused by the encoding of unsigned_long
```

Comment:
Sort order is broken when `unsigned_long` field is encoded in the `_tsid`. However, sort order is always consistent. Is there an easy fix for this? Can this be a problem we can live with?

Reply:
I think so long as its consistent it's ok - though we should probably ask around a bit. I wonder what the sort order is for signed longs right now. I'll bet negative comes after positive because we're sorting bytes not numbers.